targets/TARGET_NUVOTON/TARGET_M480/crypto/crypto-misc.cpp

```diff
@@ -119,44 +139,49 @@ void crypto_zeroize32(uint32_t *v, size_t n)
     }
 }
 
-bool crypto_aes_acquire(void)
+bool crypto_aes_acquire(bool blocking)
 {
-    return crypto_submodule_acquire(&crypto_aes_avail);
+    return crypto_aes_mutex->trylock_for(blocking ? osWaitForever : 0);
 }
 
 void crypto_aes_release(void)
 {
-    crypto_submodule_release(&crypto_aes_avail);
+    crypto_aes_mutex->unlock();
 }
 
-bool crypto_des_acquire(void)
+bool crypto_des_acquire(bool blocking)
 {
-    return crypto_submodule_acquire(&crypto_des_avail);
+    return crypto_des_mutex->trylock_for(blocking ? osWaitForever : 0);
 }
 
 void crypto_des_release(void)
 {
-    crypto_submodule_release(&crypto_des_avail);
+    crypto_des_mutex->unlock();
 }
 
-bool crypto_sha_acquire(void)
+bool crypto_sha_acquire(bool blocking)
 {
-    return crypto_submodule_acquire(&crypto_sha_avail);
+    if (blocking) {
+        while (core_util_atomic_flag_test_and_set(&crypto_sha_atomic_flag));
```

Comment:
No, you mustn't ever do this. And you don't, so take out the `blocking` case.

Reply:
@kjbracey-arm As above, `blocking` parameter has removed.